### PR TITLE
Bugfix/analyses EE2

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

- [ ] Describe testing already performed for this Pull Request:</br>

Tested both the stats and plots with these EE2 changes:

- Ensured that only COMIN and COMOUT are the only two definitions in the driver scripts
- Set COMIN and COMOUT properly in the jobs script (and COMIN is not used to define COMOUT)
- Items like COMOUTplots no longer in driver script.
- VDATE removed in driver script
- KEEPDATA and SENDMAIL are defined in driver script and set to YES
- SENDMAIL statements wrapped around mail messages
- SENDDBN alert for plot tarball added
- modulefiles updated to remove pathname

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Please test all stats and plots scripts.

For stats, please be sure to add -v cyc=00 in your code (or whatever cycle you choose to use
For plots, please set COMIN to emc.vpppg parallel.

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

ASAP

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ cd $PBS_O_WORKDIR
 
 export model=evs
 
-export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS
+export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS_analyses_EE2/EVS
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh]: Can you revert this HOMEevs definition?

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_ru_grid2obs_stats.sh [hunk 2]
```diff
@@ -5,12 +5,12 @@
...
 #PBS -A VERF-DEV
 #PBS -l walltime=00:15:00
-#PBS -l select=1:ncpus=1:mem=2GB
+#PBS -l select=1:ncpus=1:mem=10GB
 #PBS -l debug=true
 
 export model=evs
 
-export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS
+export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS_analyses_EE2/EVS
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_ru_grid2obs_stats.sh]: Can you revert this HOMEevs definition?

File: dev/drivers/scripts/analyses/stats/jevs_analyses_urma_grid2obs_stats.sh [hunk 3]
```diff
@@ -6,12 +6,12 @@
...
 #PBS -A VERF-DEV
 #PBS -l walltime=00:30:00
-#PBS -l select=1:ncpus=1:mem=2GB
+#PBS -l select=1:ncpus=1:mem=10GB
 #PBS -l debug=true
 
 export model=evs
 
-export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS
+export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS_analyses_EE2/EVS
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/analyses/stats/jevs_analyses_urma_grid2obs_stats.sh]: Can you revert this HOMEevs definition? ... and all the rest of the drivers

File: jobs/JEVS_ANALYSES_PLOTS [hunk 4]
```diff
@@ -56,10 +57,10 @@ export RUN=${RUN:-atmos}
 
 echo $COMPATH
 
-export COMIN=${COMIN:-$(compath.py ${envir}/com/evs/${evs_ver}/)}
-export COMINanl=${COMINanl:-$(compath.py ${envir}/com/evs/${evs_ver}/stats/${COMPONENT})}
-export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}}
-export COMOUTplots=${COMOUTplots:-${COMOUT}/${RUN}.${VDATE}}
+export COMIN=${COMIN:-$(compath.py ${envir}/com/evs/${evs_ver})}
+export COMINanl=${COMINanl:-${COMIN}/stats/${COMPONENT}}
+export COMOUT=${COMOUT:-$(compath.py ${envir}/com/evs/${evs_ver})}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_ANALYSES_PLOTS]: This should be changed to:
```
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}
```

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh (hunk 1 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_grid2obs_stats.sh]: Nerts, I thought I took care of that!

Perry

File: dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_ru_grid2obs_stats.sh (hunk 2 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/stats/jevs_analyses_rtma_ru_grid2obs_stats.sh]: See above.

File: dev/drivers/scripts/analyses/stats/jevs_analyses_urma_grid2obs_stats.sh (hunk 3 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/stats/jevs_analyses_urma_grid2obs_stats.sh]: Also above.

File: jobs/JEVS_ANALYSES_STATS [hunk 5]
```diff
@@ -61,10 +62,10 @@ fi
 export COMIN=${COMIN:-$(compath.py -o ${envir}/com/${NET}/$evs_ver)}
 export COMINfcst=${COMINfcst:-$(compath.py ${envir}/com/${modsys}/${mod_ver})}
 export COMINobs=${COMINobs:-$(compath.py ${envir}/com/obsproc/${obsproc_ver})}
-export COMOUT=${COMOUT:-$(compath.py ${NET}/${evs_ver}/${STEP}/${COMPONENT}/${RUN})}
+export COMOUT=${COMOUT:-$(compath.py ${NET}/${evs_ver})}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_ANALYSES_STATS]: This should be changed to:
```
export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}
```

File: jobs/JEVS_ANALYSES_STATS [hunk 6]
```diff
@@ -61,10 +62,10 @@ fi
 export COMIN=${COMIN:-$(compath.py -o ${envir}/com/${NET}/$evs_ver)}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_ANALYSES_STATS]: And I think this one should be no `-o` if you are going to use `${envir}/com/`.

File: jobs/JEVS_ANALYSES_STATS (hunk 5 above)
Comment (PerryShafran-NOAA) [reply jobs/JEVS_ANALYSES_STATS]: Done.

File: jobs/JEVS_ANALYSES_PLOTS [hunk 7]
```diff
@@ -56,10 +57,10 @@ export RUN=${RUN:-atmos}
 
 echo $COMPATH
 
-export COMIN=${COMIN:-$(compath.py ${envir}/com/evs/${evs_ver}/)}
-export COMINanl=${COMINanl:-$(compath.py ${envir}/com/evs/${evs_ver}/stats/${COMPONENT})}
-export COMOUT=${COMOUT:-${COMIN}/${STEP}/${COMPONENT}}
-export COMOUTplots=${COMOUTplots:-${COMOUT}/${RUN}.${VDATE}}
+export COMIN=${COMIN:-$(compath.py ${envir}/com/evs/${evs_ver})}
+export COMINanl=${COMINanl:-${COMIN}/stats/${COMPONENT}}
+export COMOUT=${COMOUT:-$(compath.py -o evs/${evs_ver})}
```
Comment (ShelleyMelchior-NOAA) [inline jobs/JEVS_ANALYSES_PLOTS]: Thanks! Should we make this one use ${NET} instead of evs?
`export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}`

**Comment by ShelleyMelchior-NOAA** [comment]
I have cloned your fork and checked out branch `bugfix/analyses_EE2`.

File: dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh [hunk 8]
```diff
@@ -28,7 +28,8 @@ module load prod_envir/${prod_envir_ver}
 
 export envir=prod
 export DATAROOT=/lfs/h2/emc/stmp/${USER}/evs_test/$envir/tmp
-export KEEPDATA=NO
+export KEEPDATA=YES
+export SENDMAIL= YES
```
Comment (AliciaBentley-NOAA) [inline dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh]: @PerryShafran-NOAA Super tiny, but can this extra space be deleted?

File: jobs/JEVS_ANALYSES_PLOTS (hunk 7 above)
Comment (PerryShafran-NOAA) [reply jobs/JEVS_ANALYSES_PLOTS]: Done.

File: dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh (hunk 8 above)
Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/analyses/plots/jevs_analyses_grid2obs_plots.sh]: That's good lookin' out!  I removed that space.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA It looks like SENDDBN is only set to YES right now in your j-jobs. Can you set it to NO in the dev drivers so we don't use the SENDDBN alerts?

**Comment by PerryShafran-NOAA** [comment]
@AliciaBentley-NOAA I was making that change JUST as you were making your comment.  I set it to NO in the J-job - should it be YES in the J-job and NO in the driver?

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA In your J-jobs, can you please move your "export SEND____" alerts to underneath the the text that describes these alerts? So it would look more like this instead of having these things in two separate parts of the J-job?

```
##################################################
## SAVEGES  - Copy Files From TMPDIR to $GESdir
## SENDECF  - Flag Events on ecFLOW
## SENDCOM  - Copy Files From TMPDIR to $COMOUT
## SENDDBN  - Issue DBNet Client Calls
## SENDMAIL - Send email if file is missing
##################################################
export SENDCOM=${SENDCOM:-YES}
export SENDDBN=${SENDDBN:-NO}
export SENDECF=${SENDECF:-YES}
export SENDMAIL=${SENDMAIL:-NO}
```

**Comment by PerryShafran-NOAA** [comment]
On it.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA And this one may not matter, but is the "set -x**a**" (specifically the "a") here intended? Most of your other code just has "set -x".

https://github.com/NOAA-EMC/EVS/pull/271/files#diff-4663d2e2e37a1961c162bb94e0d92f3a898cdc7301e380beac053fd61b1de1d3R6

**Comment by PerryShafran-NOAA** [comment]
I don't know?  I'll go ahead and remove the a.

**Comment by AliciaBentley-NOAA** [comment]
@ShelleyMelchior-NOAA @PerryShafran-NOAA I'm done with my review! All boxes have been checked off and turned gray under /analyses in the EE2 checklist!

**Comment by ShelleyMelchior-NOAA** [comment]
I can start setting up these test runs.

**Comment by ShelleyMelchior-NOAA** [comment]
The stats processing has been submitted.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/stats/analyses/atmos.20230929/`
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_*o.*`

**Comment by ShelleyMelchior-NOAA** [comment]
I tried to start plots processing but I am stumped. It is unhappy about `setpdy.sh`. I even tried to run it in command line to isolate the error. No luck. I have posted about this on the WCOSS2 slack channel.
```
[dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ which setpdy.sh
/apps/ops/prod/nco/core/prod_util.v2.0.8/ush/setpdy.sh
[dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ cyc=00
[dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ cycle=t${cyc}z
[dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ echo $cyc $cycle
00 t00z
[dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ setpdy.sh

-------------------------------------------------------------
-- FATAL ERROR: cycle variable not set or formatted incorrectly
-- ABNORMAL EXIT at Sat Sep 30 00:45:54 UTC 2023 on dlogin02
-------------------------------------------------------------
```

File: modulefiles/analyses/analyses_plots.sh
Comment (ShelleyMelchior-NOAA) [inline modulefiles/analyses/analyses_plots.sh:1]: line 13 has a typo. It should be `module load prod_util/${prod_util_ver}`

File: modulefiles/analyses/analyses_stats.sh
Comment (ShelleyMelchior-NOAA) [inline modulefiles/analyses/analyses_stats.sh:1]: line 15 has a typo. It should be `module load prod_util/${prod_util_ver}`

File: modulefiles/analyses/analyses_plots.sh [hunk 9]
```diff
@@ -1,9 +1,10 @@
 #!/bin/bash
-# modulefile for EVS subseasonal component
+# modulefile for EVS analyses plot component
 
-set +x
+set -x
 
-module use /apps/prod/lmodules/intel/${intel_ver}
+module load PrgEnv-intel/${PrgEnvintel_ver}
+module load intel/${intel_ver)
```
Comment (ShelleyMelchior-NOAA) [inline modulefiles/analyses/analyses_plots.sh]: Please correct this typo. Needs a } instead of ).

**Comment by ShelleyMelchior-NOAA** [comment]
> I tried to start plots processing but I am stumped. It is unhappy about `setpdy.sh`. I even tried to run it in command line to isolate the error. No luck. I have posted about this on the WCOSS2 slack channel.
> 
> ```
> [dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ which setpdy.sh
> /apps/ops/prod/nco/core/prod_util.v2.0.8/ush/setpdy.sh
> [dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ cyc=00
> [dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ cycle=t${cyc}z
> [dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ echo $cyc $cycle
> 00 t00z
> [dlogin02 /lfs/h2/emc/vpppg/noscrub/shelley.melchior]$ setpdy.sh
> 
> -------------------------------------------------------------
> -- FATAL ERROR: cycle variable not set or formatted incorrectly
> -- ABNORMAL EXIT at Sat Sep 30 00:45:54 UTC 2023 on dlogin02
> -------------------------------------------------------------
> ```

The setpdy.sh problems are resolved after tracking down some typos in the analyses plots modulefile. 
My command line usage of setpdy.sh was failing b/c I was not exporting cycle.

**Comment by ShelleyMelchior-NOAA** [comment]
The plots processing is underway. But it is encroaching on wallclock limitations:
```
83183487.cbqs01                shelley.melchi* dev             jevs_analyses_*   122325    1     1   10gb 01:00 R 00:58
```
I will re-run and restart should take over to finish it quickly. But @PerryShafran-NOAA , maybe the wallclock should be stretched to 2 hours?

This is what finished:
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/analyses/atmos.20230929`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o83183487`

Re-run logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_grid2obs_plots.o83186034`
I see lots of "RESTART" in the log file.
And now the tar file is complete:
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/analyses/atmos.20230929/
total 221404
drwxr-sr-x 2 shelley.melchior vpppg     12288 Sep 30 14:12 CEILING
drwxr-sr-x 2 shelley.melchior vpppg      4096 Sep 30 14:10 DPT
-rw-r--r-- 1 shelley.melchior vpppg 226662400 Sep 30 14:20 evs.plots.analyses.atmos.grid2obs.last31days.v20230929.tar
drwxr-sr-x 2 shelley.melchior vpppg      4096 Sep 30 14:11 GUST
drwxr-sr-x 2 shelley.melchior vpppg     12288 Sep 30 14:12 TCDC
drwxr-sr-x 2 shelley.melchior vpppg      4096 Sep 30 14:10 TMP
drwxr-sr-x 2 shelley.melchior vpppg     12288 Sep 30 14:11 VIS
drwxr-sr-x 2 shelley.melchior vpppg      4096 Sep 30 14:10 WIND
```
CONCLUSION: analyses plots looks good, but please fix the modulefiles typos and extend the walltime. It looks like the re-run only needed 4.5 more minutes to do the copying and tar file creation.

**Comment by AliciaBentley-NOAA** [comment]
@ShelleyMelchior-NOAA Thanks for finding those typos! Hopefully those can be fixed soon!

@PerryShafran-NOAA  I recommend increasing the wallclock time more than needed (perhaps by 30 more minutes to 1 hour) so that jobs still finish in the emc.vpppg parallel even when the machine is super slow. Thanks.

File: modulefiles/analyses/analyses_plots.sh
Comment (PerryShafran-NOAA) [reply modulefiles/analyses/analyses_plots.sh:1]: Fixed.

File: modulefiles/analyses/analyses_stats.sh
Comment (PerryShafran-NOAA) [reply modulefiles/analyses/analyses_stats.sh:1]: Fixed.

File: modulefiles/analyses/analyses_plots.sh (hunk 9 above)
Comment (PerryShafran-NOAA) [reply modulefiles/analyses/analyses_plots.sh]: Fixed.

**Comment by PerryShafran-NOAA** [comment]
I extended the walltime from 1 hr to 2 hrs.

**Comment by PerryShafran-NOAA** [comment]
The stats and plot output looks good.  

I am also seeing the module loads of PrgEnv-intel and intel.

I am also seeing that the plot log file is included in the output, which is good.  

I think that we are good here.  

Perry

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
The testing in this PR produced expected results. This PR is approved for merge. @PerryShafran-NOAA , please do one more sync fork.

**Pull request merged**